### PR TITLE
Added autoscale option support

### PR DESCRIPTION
**Pull request by geruchvitaliy**
Added autoscale option support into collection and database options, so contributor can choose which scale model to use - Auto or Manual.

Manual scale model used as default, so package is compatible with previous versions.

### CODE REVIEW

File: src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs [hunk 1]
```diff
@@ -11,9 +11,11 @@ namespace SimpleEventStore.CosmosDb
 {
     internal class AzureCosmosDbStorageEngine : IStorageEngine
     {
+        private const int DefaultAutoscaleMaxThroughput = 4000;
```
Comment (asosMikeGore) [inline src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs:14]: I'm not sure we should be setting a default throughput - while the setting is the minimum supported value, I'd rather have people think about the offer levels they want to use so the cost and throughput trade-offs are taken into consideration.

File: src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs [hunk 2]
```diff
@@ -124,14 +126,31 @@ FROM e
...
+            if (_databaseOptions.UseAutoscale)
+            {
+                if (_databaseOptions.DatabaseRequestUnits != null)
+                {
+                    return _client.CreateDatabaseIfNotExistsAsync(_databaseName,
+                        ThroughputProperties.CreateAutoscaleThroughput((int)_databaseOptions.DatabaseRequestUnits));
+                }
+                else
+                {
+                    return _client.CreateDatabaseIfNotExistsAsync(_databaseName,
```
Comment (asosMikeGore) [inline src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs:138]: If we removed the default offer support then we could remove the logic in the else block

**Comment by asosMikeGore** [comment]
@geruchvitaliy wondering what you think to having the collection & db initialization in the code base compared to leaving this up to an infrastructure as code tool e.g. Bicep?  When we initially put together the library there wasn't support in ARM templates to manage collections which was one of the reasons we added it in.

Just thinking whether this provides a way of simplifying the library - the main risk would be though that collections are added that don't have the correct schema. However, we could check for this, rather than creating the collections etc. I'll ask some of our engineers for feedback here too.

**Comment by afscrome** [comment]
I'd +1 removing initalisation code from the library.  Several times now we've had to play catch up to new cosmos features (#37, this PR).  Especially with the recent remove of the stored procedure this is easier than ever.

File: src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs (hunk 1 above)
Comment (geruchvitaliy) [reply src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs:14]: This parameter will be used only when user will not specify the custom collection/database throughput and set the 'UseAutoscale' flag to be True. It's minimum number which can be used for Auto scale mode, so we can not leave it empty or use 0.

If we will not do that, there will be raised an error when 'UseAutoscale' flag will be True:

`Microsoft.Azure.Cosmos.CosmosException: 'Response status code does not indicate success: BadRequest (400); Substatus: 0; ActivityId: c9178727-9310-4d01-b0ff-965406a7e9bb; Reason: (Message: {"Errors":["The offer should have valid throughput values between 4000 and 1000000 inclusive in increments of 1000."]}`

**Comment by afscrome** [comment]
That said, it may still be useful to have some helpers to do basic initialisation of a collection - this could particularly be useful for local dev environments using hte Cosmos emulator.

**Comment by geruchvitaliy** [comment]
Removing the initialization code, I think, it's a different topic and it is out of scope this PR.

Yes, I agree, the initialization code need to be removed, the package should simply take existing resources and does read or append events.

**Comment by asosMikeGore** [comment]
@geruchvitaliy I agree that removing the initialization code is outside the scope of the PR. @afscrome good point about the local dev experience.

I suspect a sensible middle ground is we leave the ability to initialize the collections with the expected schema & indexes, but remove any form of scaling config.  That then makes it easy for local dev and then in deployments to test & prod environments rely on infrastructure as code tools to configure the collections, scale settings etc.

**Pull request closed without merging**